tests/unittests/links.cpp

```diff
@@ -154,6 +158,64 @@ TEST_CASE("Link basics", "[links]") {
   }
 }
 // NOLINTEND(clang-analyzer-cplusplus.NewDeleteLeaks)
+TEST_CASE("Links associative containers", "[links][hash]") {
+  ExampleHit hit1, hit2;
+  ExampleCluster cluster1, cluster2;
+  TestMutL link1, link2, link3, link4;
+  link1.set(hit1);
+  link1.set(cluster1);
+  link1.setWeight(1.0);
+  link2.set(hit2);
+  link2.set(cluster2);
+  link2.setWeight(2.0);
+  link3.set(hit1);
+  link3.set(cluster2);
+  link3.setWeight(3.0);
+  link4.set(hit1);
+  link4.set(cluster1);
+  link4.setWeight(4.0);
+
+  std::set<TestL> linkSet;
+  linkSet.insert(link1);
+  linkSet.insert(link2);
+  linkSet.insert(link2);
```

Comment:
I think this would be more explicit by using `auto link5 = link2` and then `linkSet.insert(link5)`. We would then also show case that its the underlying object that matters not the handle that we use. Similar for the cases with `map` below.